app/controllers/proposals_controller.rb

```diff
@@ -98,6 +99,13 @@ def proposal_params
                                      speakers_attributes: [:bio, :user_id, :id])
   end
 
+  def require_invite
+    unless current_user.invitations.where(state: 'pending').includes(proposal_id: @proposal.id)
+      redirect_to root_path
+      flash[:danger] = 'You are not an invited speaker for the proposal you are trying to access.'
+    end
+  end
+
```

Comment:
If I'm reading this right, a user could only access the show action if they have a pending invitation. Wouldn't we also need to allow users who have accepted an invitation to see the show page?